### ISSUE TITLE
IDE1006 Naming rule for local variables does not work

### ISSUE THREAD

**Issue by Mr1008** [dotnet/roslyn#80696]
**Version Used**: .NET SDK 10.0.100-rc.1.25451.107

**Steps to Reproduce**:

1. Set following rule in .editorconfig file:
```ini
dotnet_naming_rule.locals_should_be_camel_case = error
dotnet_naming_rule.locals_should_be_camel_case.symbols = locals
dotnet_naming_rule.locals_should_be_camel_case.style = camel_case

dotnet_naming_symbols.locals.applicable_kinds = local
dotnet_naming_symbols.locals.applicable_accessibilities = local

dotnet_naming_style.camel_case.capitalization = camel_case

dotnet_diagnostic.IDE1006.severity = error
```
2. try to declare such local variable in a method:
```csharp
var ThisIsTheTest = 1;
```
3. Observe the analyzer does not report issue with name being PascalCase instead of camelCase.

**Diagnostic Id**:

IDE1006

**Expected Behavior**:
The naming rules should be applied to local variables.

**Actual Behavior**:
Local variables are not detected, despite being configured correctly in .editorconfig.

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/18121
- https://github.com/dotnet/roslyn/issues/28088
- https://github.com/dotnet/roslyn/issues/13249

**Comment by russellfoster** [comment]
Seeing this problem as well. What used to fail reliably in VS 2019 builds (IDE1006 errors) now passes in VS 2022, which means we are now seeing the violations sneak into our codebase. Can we at least get a confirmation of this regression?
